docs/source/community-model-flavors.rst

```diff
@@ -0,0 +1,365 @@
+
+Documentation
+^^^^^^^^^^^^^
+
+Usage examples for all flavors and the API reference can be found in the package
+`documenation <https://mlflavors.readthedocs.io/en/latest/index.html>`_.
+
+
+Installation
+^^^^^^^^^^^^
+
+Installing from PyPI:
+
+.. code-block:: bash
+
+   $ pip install mlflavors
+
+Quickstart
+^^^^^^^^^^
+
+This example trains a `PyOD <https://github.com/yzhao062/pyod>`_ KNN outlier detection
+model using a synthetic dataset. A new MLflow experiment is created to log the evaluation
+metrics and the trained model as an artifact and anomaly scores are computed loading the
+trained model in native flavor and ``pyfunc`` flavor. Finally, the model is served
+for real-time inference using a local endpoint.
+
+Saving the model as an MLflow artifact
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    import json
+
+    import mlflow
+    import pandas as pd
+    from pyod.models.knn import KNN
+    from pyod.utils.data import generate_data
+    from sklearn.metrics import roc_auc_score
+
+    import mlflavors
+
+    ARTIFACT_PATH = "model"
+
+    with mlflow.start_run() as run:
+        contamination = 0.1  # percentage of outliers
+        n_train = 200  # number of training points
+        n_test = 100  # number of testing points
+
+        X_train, X_test, _, y_test = generate_data(
+            n_train=n_train, n_test=n_test, contamination=contamination
+        )
+
+        # Train kNN detector
+        clf = KNN()
+        clf.fit(X_train)
+
+        # Evaluate model
+        y_test_scores = clf.decision_function(X_test)
+
+        metrics = {
+            "roc": roc_auc_score(y_test, y_test_scores),
+        }
+
+        print(f"Metrics: \n{json.dumps(metrics, indent=2)}")
+
+        # Log metrics
+        mlflow.log_metrics(metrics)
+
+        # Log model using pickle serialization (default).
+        mlflavors.pyod.log_model(
+            pyod_model=clf,
+            artifact_path=ARTIFACT_PATH,
+            serialization_format="pickle",
+        )
+        model_uri = mlflow.get_artifact_uri(ARTIFACT_PATH)
+
+    # Print the run id wich is used below for serving the model to a local REST API endpoint
+    print(f"\nMLflow run id:\n{run.info.run_id}")
+
+Loading the model from MLflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Make a prediction loading the model from MLflow in native format:
+
+.. code-block:: python
+
+    loaded_model = mlflavors.pyod.load_model(model_uri=model_uri)
+    print(loaded_model.decision_function(X_test))
+
+Make a prediction loading the model from MLflow in ``pyfunc`` format:
+
+.. code-block:: python
+
+    loaded_pyfunc = mlflavors.pyod.pyfunc.load_model(model_uri=model_uri)
+
+    # Create configuration DataFrame
+    predict_conf = pd.DataFrame(
+        [
+            {
+                "X": X_test,
+                "predict_method": "decision_function",
+            }
+        ]
+    )
+
+    print(loaded_pyfunc.predict(predict_conf)[0])
```

Comment:
The interface looks a bit strange, ususally `pyfunc_model.predict` interface only accepts the inference dataset, but here your interface it accepts an object containing both inference dataset and config setting.

Usually we should save related config setting when saving model, and when loading model as a pyfunc model, we read the setting value and set it automatically in for the loaded pyfunc model.

The issue does not block the doc PR but I suggest to improve the interface to make it consistent with other ML flavor interfaces.

Reply:
Thanks for your suggestion @WeichenXu123 

Actually, I do not see an inconsistency here with MLflow built-in flavors. In fact, the same approach has been followed for example in the MLflow built-in [Pmdarima](https://mlflow.org/docs/latest/_modules/mlflow/pmdarima.html) (see the **_PmdarimaModelWrapper**) flavor - it also exposes the inference dataset plus additional inference API parameters in the pyfunc configuration dataframe. The same applies for the [MLflow custom flavor example](https://mlflow.org/docs/latest/models.html#example-creating-a-custom-sktime-flavor) for sktime library (see the **_SktimeModelWrapper**).

Just out of curiosity, can you point me to a built-in flavor example where the parameters of the inference API are saved together with the model?